xaitk_saliency/utils/masking.py

```diff
@@ -1,12 +1,64 @@
-from typing import List
+from typing import List, Tuple
 
 import numpy as np
 import PIL.Image
 
 
+def generate_block_masks(
+    window_size: int,
+    stride: int,
+    image_size: Tuple[int, int],
+) -> np.ndarray:
+    """
+    Generates a number of mask images with each masking a distinct square area
+    within an input image shape.
+
+    An amount of masks generated will be equal to the number of full
+    `window_size x window_size` blocks we can fit into the given shape with the
+    given stride.
+    If the input shape is not evenly divisible by window size and stride then
+    there may be pixels on teh right and bottom that are never masked.
+
+    :param window_size: the block window size (with value 0, other areas
+        with value 1)
+    :param stride: the sliding step
+    :param image_size: the mask size which should be the same to the image
+        size
+
+    :return: the sliding window style masks
+    """
+    rows = np.arange(0 + stride - window_size, image_size[0], stride)
```

Comment:
Not 100% aware of the logic of the sliding window occlusions, but wanted to confirm that the following (apparent) behaviors were expected:
a) if stride > window_size, then the points along the perimeter of the masks [where i, j < (stride - window_size)] will not be 0 in any of the masks
b) if stride < window_size, then the set of points where r and c are both negative will all produce identical masks [ranging from (0, 0) to (window_size, window_size)]

I might be wrong, but I can double check this a bit more closely in case it's of interest

Reply:
If I understand the map generation process correctly, I'm curious if undersampling/oversampling the boundaries could potentially lead to saliency clustering or dispersing towards the edges.

Reply:
I'm not sure I'm fully parsing your question. However, I would say that this is a separate issue that we should address in a separate branch/PR after getting this initial implementation in.

Reply:
basically I'm just asking about how the ranges set in lines 30/31 affect the masks set in line 54

Reply:
@nitesh201 I believe you are right. In condition, a) boundary pixels might not be masked and I expect this would cause the saliency values to be focused around the centre of the image.

Reply:
Main point being that the statement "Generates a number of mask images with each masking a distinct square." seems to be violated under certain conditions

Reply:
:+1: I'm going to "resolve" this for now since this is appropriate for a follow-on branch/PR. This one is more simply introducing the structure and carrying the previous Phase2 implementation forward. Thank you for catching this though!